src/chat.rs

```diff
@@ -4844,10 +4880,64 @@ async fn set_contacts_by_addrs(context: &Context, id: ChatId, addrs: &[String])
     Ok(())
 }
 
+/// Set chat contacts by their fingerprints creating the corresponding contacts if necessary.
+///
+/// `fingerprint_addrs` is a list of pairs of fingerprint and address.
+async fn set_contacts_by_fingerprints(
+    context: &Context,
+    id: ChatId,
+    fingerprint_addrs: &[(String, String)],
+) -> Result<()> {
+    let chat = Chat::load_from_db(context, id).await?;
+    ensure!(
+        chat.is_encrypted(context).await?,
+        "Cannot add PGP-contacts to unencrypted chat {id}"
+    );
+    ensure!(
+        chat.typ == Chattype::Broadcast,
+        "{id} is not a broadcast list",
+    );
```

Comment:
This problem already existed in `set_contacts_by_addrs()`, i.e. it's not a new problem, but still, I must ask:

Why can this only be used for broadcast lists?

The rest of the codebase doesn't look like this function is only called for broadcast lists:
- The UI calls `add_contact_to_chat()`, which calls `add_contact_to_chat_ex()`, which calls `sync_contacts()` (except if it's a promoted group, but for unpromoted groups it's called), which adds a `SyncAction::Set(Pgp)?Contacts`
- This causes a call to `set_contacts_by_(addr|fingerprints)()` on the other device

If the condition was `chat.typ == Chattype::Broadcast || !chat.is_promoted()`, then I would understand it.